### PR TITLE
6.11 Add dalli gem and fragment caching

### PR DESCRIPTION
**Pull request by nelantone**

### CODE REVIEW

File: app/helpers/products_helper.rb [hunk 1]
```diff
@@ -1,2 +1,7 @@
 module ProductsHelper
-end
+  def cache_key_for_products
+    count          = Product.count
+    max_updated_at = Product.maximum(:updated_at).try(:utc).try(:to_s, :number)
```
Comment (wazery) [inline app/helpers/products_helper.rb:4]: :+1:

File: app/views/products/_product.html.erb [hunk 2]
```diff
@@ -1,14 +1,16 @@
...
-  <%= link_to 'Edit', edit_product_path(product) %>
-  <%= link_to 'Destroy', product, method: :delete, data: { confirm: 'Are you sure?' } %>
-  <br>
-</div> <!-- col-sm-2 col-md-4 -->
\ No newline at end of file
+<% cache product do %>
+  <div class="col-sm-2 col-md-4">
+    <br>
+    <%= product.name %><br>
+    <% if product.image_url.nil? %>
```
Comment (wazery) [inline app/views/products/_product.html.erb]: no need for nil?, you can remove it and reverse the code blocks

**Review by wazery — APPROVED** [review]
Good job, LGTM 👍  😄

File: app/views/products/_product.html.erb (hunk 2 above)
Comment (nelantone) [reply app/views/products/_product.html.erb]: Hoh! Yes o-O ^^.
Thanks

**Comment by nelantone** [comment]
:ship: ing! :+1:

**Pull request merged**